raft/raft.go

```diff
@@ -513,12 +503,17 @@ func (r *raft) sendHeartbeat(to uint64, ctx []byte) {
 // bcastAppend sends RPC, with entries to all peers that are not up-to-date
 // according to the progress recorded in r.prs.
 func (r *raft) bcastAppend() {
-	for id := range r.prs {
-		if id == r.id {
-			continue
+	f := func(prs map[uint64]*Progress) {
```

Comment:
make this a method `forEachProgress(prs [], prs[], f())`? so we do not need to repeat this quite a few times.